pkg/bgpv1/agent/controller.go

```diff
@@ -44,12 +46,31 @@ func (plf policyListerFunc) List() ([]*v2alpha1api.CiliumBGPPeeringPolicy, error
 	return plf()
 }
 
+type ipPoolLister interface {
+	List() ([]*v2alpha1api.CiliumPodIPPool, error)
+}
+
+type ipPoolListerFunc func() ([]*v2alpha1api.CiliumPodIPPool, error)
+
+func (iplf ipPoolListerFunc) List() ([]*v2alpha1api.CiliumPodIPPool, error) {
+	return iplf()
+}
+
 // Controller is the agent side BGP Control Plane controller.
 //
 // Controller listens for events and drives BGP related sub-systems
 // to maintain a desired state.
 type Controller struct {
 	LocalNodeStore *node.LocalNodeStore
+	// IPPoolResource provides a store of cached pod ip pools and allows us to observe changes
+	// to the objects in its store.
+	IPPoolResource resource.Resource[*v2alpha1api.CiliumPodIPPool]
+	// IPPoolLister is an interface which allows for the listing of the CiliumPodIPPool resource.
+	IPPoolLister ipPoolLister
```

Comment:
Do we still need `IPPoolResource` and `IPPoolLister` in the controller now that we use `BGPCPResourceStore[*v2alpha1api.CiliumPodIPPool]` in the reconciler?

Reply:
Both are unneeded and have been removed in commit 572b94b.